File: Python/Programs-based-on-Numbers/24-HCF-of-two-numbers.py
num1 = int(input("Enter first number: "))
num2 = int(input("Enter Second number: "))

while num2 != 0:
    num1, num2 = num2, num1 % num2

print("\nHCF is:", num1)


# Using for loop 
# hcf = 1
# for i in range(1, (min(num1,num2)+1)):
#     if (num1%i == 0) and (num2%i == 0):
#         hcf = i

# print("\nHCF is:", hcf)

File: Java/2-Programs-based-on-Numbers/24-HCF-of-two-numbers.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter first number:");
        int num1 = getInput.nextInt();

        System.out.println("\nEnter second number:");
        int num2 = getInput.nextInt();

        while (num2!=0) {
            int temp = num2;
            num2 = num1 % num2;
            num1 = temp;
        }
        System.out.println("\nHCF is: " + num1);
        getInput.close();
    }
}